html/includes/modal/edit_alert_transport.inc.php

```diff
@@ -0,0 +1,246 @@
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+use LibreNMS\Authentication\Auth;
+use LibreNMS\Config;
+
+if (Auth::user()->hasGlobalAdmin()) {
+?>
+<!--Modal for adding or updating an alert transport -->
+    <div class="modal fade" id="edit-alert-transport" tabindex="-1" role="dialog"
+         aria-labelledby="Edit-transport" aria-hidden="true">
+        <div class="modal-dialog modal-lg">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                    <h5 class="modal-title" id="Edit-transport">Alert Transport :: <a href="https://docs.librenms.org/Alerting/">Docs <i class="fa fa-book fa-1x"></i></a> </h5>
+                </div>
+                <div class="modal-body">
+                    <form method="post" role="form" id="transports" class="form-horizontal transports-form">
+                        <input type="hidden" name="transport_id" id="transport_id" value="">
+                        <input type="hidden" name="type" id="type" value="alert-transports">
+                        <div class='form-group' title="The description of this alert transport.">
+                            <label for='name' class='col-sm-3 col-md-2 control-label'>Transport name: </label>
+                            <div class='col-sm-9 col-md-10'>
+                                <input type='text' id='name' name='name' class='form-control validation' maxlength='200' required>
+                            </div>
+                        </div>
+                        <div class="form-group" title="The type of transport.">
+                            <label for='transport-choice' class='col-sm-3 col-md-2 control-label'>Transport type: </label>
+                            <div class="col-sm-3">
+                                <select name='transport-choice' id='transport-choice' class='form-control'>
+                                    <option value="mail-form" selected>Mail</option>
+                                    <option value="ciscospark-form">Cisco Spark</option>
+                                    <!--Insert more transport type options here has support is added. Value should be: [transport_name]-form -->
+                                </select>
+                            </div>
+                        </div>
+                        <div class="form-group" title="The transport is default.">
+                            <label for="default" class="col-sm-3 col-md-2 control-label">Default Alert: </label>
+                            <div class="col-sm-2">
+                                <input type="checkbox" name="is_default" id="is_default">
+                            </div>
+                        </div>
+                    </form>
+<?php
+$transports = Config::get('alert.transports');
```

Comment:
So this only gets transports we've added to the DB. Not all of them will have db entries so we need to find another way to loop through the actual transports.